tests/mypy/test_mypy.py

```diff
@@ -0,0 +1,72 @@
+import importlib
+import os
+from pathlib import Path
+
+import pytest
+from mypy import api
+
+try:
+    import typing_extensions
+except ImportError:
+    typing_extensions = None
+
+# This ensures mypy can find the test files, no matter where tests are run from:
+os.chdir(Path(__file__).parent.parent.parent)
+
+# You can change the following variable to True during development to overwrite expected output with generated output
+GENERATE = False
+
+cases = [
+    ('mypy-plugin.ini', 'plugin_success.py', None),
+    ('mypy-plugin.ini', 'plugin_fail.py', 'plugin-fail.txt'),
```

Comment:
are you sure these are really easier as separate files? Surely easier to have the expected output right here with the test even if some are very verbose?

Reply:
It's *extremely* hard to iterate on it without a mechanism for generating the expected output. By putting it in a file, I can set `GENERATE=True` above and it will write the file with the actual output, and I can look at the git diffs to see if things are as desired.

I initially tried having the results in-line here, but the problem is that a one line changes *all* the line numbers. And the line numbers actually do matter, so I think we shouldn't just strip them from the expected output. But then every time I tweak a line I had to manually compare all the output lines and fix the line numbers. Then I'd make another minor change and have to do it all again.

I think there might be better ways to manage mypy plugin tests (might be good to look at how the sqlalchemy or django plugins do this). But I thought they might ultimately rely on calling mypy on a separate file for each test, and I was specifically trying to minimize the amount of time spent executing mypy tests during normal test runs, and I think there is a substantial amount of overhead for each file analyzed. I tried to get all the tests into as few files as possible to keep the tests running fast.